### PR TITLE
Remove the default bound from Packable::Visitor

### PR DESCRIPTION
**Pull request by DaughterOfMars**
## Description

This PR removes the `Default` bound from the `Packable::Visitor`. As a consequence, the visitor must be optional in the trait. Users can now check whether the visitor is `Some` to validate their data, so the `const VERIFY` generic is no longer useful, thus it has been removed.

### CODE REVIEW

File: packable/packable-derive/src/fragments.rs [hunk 1]
```diff
@@ -32,21 +32,39 @@ impl Fragments {
...
+                    quote! {
+                        if let Some(visitor) = visitor {
+                            #verify_with(&#field_ident, visitor).map_err(#crate_name::error::UnpackError::from_packable)?;
+                        }
+                    }
+                } else {
+                    quote! {
+                        if visitor.is_some() {
+                            #verify_with(&#field_ident).map_err(#crate_name::error::UnpackError::from_packable)?;
+                        }
```
Comment (Thoralf-M) [inline packable/packable-derive/src/fragments.rs:47]: Can you maybe add some comment in the code why this is done like this?

**Review by thibault-martinez — CHANGES_REQUESTED** [review]
Tests are failing

File: packable/packable-derive/src/fragments.rs (hunk 1 above)
Comment (DaughterOfMars) [reply packable/packable-derive/src/fragments.rs:47]: I mean...it's in the derive crate so I don't know if there's any point.

File: packable/packable-derive/CHANGELOG.md [hunk 2]
```diff
@@ -19,6 +19,13 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Security -->
 
+## 0.10.0 - 2024-02-08
```
Comment (thibault-martinez) [inline packable/packable-derive/CHANGELOG.md:22]:
```suggestion
## 0.10.0 - 2024-02-09
```

File: packable/packable/CHANGELOG.md [hunk 3]
```diff
@@ -19,6 +19,18 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Security -->
 
+## 0.11.0 - 2024-02-08
```
Comment (thibault-martinez) [inline packable/packable/CHANGELOG.md:22]:
```suggestion
## 0.11.0 - 2024-02-09
```

File: packable/packable/src/packable/prefix/vec.rs [hunk 4]
```diff
@@ -110,12 +110,12 @@ where
...
     #[inline]
-    fn unpack<U: Unpacker, const VERIFY: bool>(
+    fn unpack<U: Unpacker>(
         unpacker: &mut U,
-        visitor: &Self::UnpackVisitor,
+        visitor: Option<&Self::UnpackVisitor>,
     ) -> Result<Self, UnpackError<Self::UnpackError, U::Error>> {
         // The length of any dynamically-sized sequence must be prefixed.
-        let len = B::unpack::<_, VERIFY>(unpacker, &())
+        let len = B::unpack_inner(unpacker, visitor)
```
Comment (thibault-martinez) [inline packable/packable/src/packable/prefix/vec.rs:118]: Can you ELI5 me when it's possible to call `unpack` and when not?

File: packable/packable/tests/common/mod.rs [hunk 5]
```diff
@@ -52,15 +52,15 @@ where
...
+    let unpacked = P::unpack_bytes_verified(vec.as_slice(), &()).unwrap();
     assert_eq!(packable, &unpacked);
 
     // Tests for `Read` and `Write`
 
     let mut packer = IoPacker::new(Vec::new());
     packable.pack(&mut packer).unwrap();
     let mut unpacker = IoUnpacker::new(packer.as_slice());
-    let unpacked = P::unpack::<_, true>(&mut unpacker, &()).unwrap();
+    let unpacked = P::unpack(&mut unpacker, None).unwrap();
```
Comment (Thoralf-M) [inline packable/packable/tests/common/mod.rs:63]: Before this had `true`, wouldn't the equivalent now be `Some(())`?

Comment (DaughterOfMars) [reply packable/packable/tests/common/mod.rs:63]: actually it would be `unpack_inner`. I missed it, thanks

File: packable/packable/src/packable/prefix/vec.rs (hunk 4 above)
Comment (Thoralf-M) [reply packable/packable/src/packable/prefix/vec.rs:118]: The same as what?

File: packable/packable/tests/common/mod.rs (hunk 5 above)
Comment (DaughterOfMars) [reply packable/packable/tests/common/mod.rs]: actually,  I was wrong in this case as the visitor type is always `()`, but anyway I fixed it

File: packable/packable/src/packable/prefix/vec.rs (hunk 4 above)
Comment (DaughterOfMars) [reply packable/packable/src/packable/prefix/vec.rs:118]: the visitor type of the type you are unpacking

File: packable/packable-derive/CHANGELOG.md [hunk 6]
```diff
@@ -19,6 +19,13 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Security -->
 
+## 0.10.0 - 2024-02-09
+
+### Changed
+
+- Updated to new `Packable` trait.
```
Comment (thibault-martinez) [inline packable/packable-derive/CHANGELOG.md:26]:
```suggestion
- Updated to new `Packable` trait;
```

**Review by thibault-martinez — APPROVED** [review]

**Review by Thoralf-M — APPROVED** [review]
